include/eve/detail/function/simd/x86/friends.hpp

```diff
@@ -163,7 +173,20 @@ namespace eve::detail
     }
     else
     {
-      return bit_cast(v.bits() == w.bits(), as(v));
+      constexpr auto c = categorize<wide<T, N, ABI>>();
+
+      if constexpr( ( c == category::int64x4 || c == category::uint64x4 ||
+                      c == category::int32x8 || c == category::uint32x8 )
+                  && current_api == avx
+                  )
+      {
+        // AVX has a bad == for those integers so we use XOR
+        return bit_cast(~v.bits() ^ w.bits(), as(v));
+      }
```

Comment:
I suspect that `~v.bits()` is `andnot_zero`.
I suspect you can do:

`_mm256_cmp_ps(v, w, _CMP_UNORD_Q)` (well pd for 64 bit)

Since true is Nan. Judging by the description of `_mm_cmpunord_pd` this is exactly what we need.